src/commands/aksRetinaCapture/aksRetinaCapture.ts

```diff
@@ -0,0 +1,133 @@
+
+    const extension = getExtension();
+    if (failed(extension)) {
+        vscode.window.showErrorMessage(extension.error);
+        return;
+    }
+
+    const kubeConfigFile = await tmpfile.createTempFile(clusterInfo.result.kubeconfigYaml, "yaml");
+    console.log(`Kubeconfig file: ${kubeConfigFile}`);
+
+    // Get all Nodes For this Cluster
+    const nodelistResult = await invokeKubectlCommand(
+        kubectl,
+        kubeConfigFile.filePath,
+        `get nodes -o jsonpath="{range .items[*]}{.metadata.name}{' '}{end}"`,
+    );
+
+    if (failed(nodelistResult)) {
+        vscode.window.showErrorMessage(`Failed to capture the cluster: ${nodelistResult.error}`);
+        return;
+    }
+    const nodesList = nodelistResult.result.stdout.trim().split(" ").map((nodeName) => ({ label: nodeName }));
+
+    // Pick a cluster from group to compare with
+    const nodeNamesSelected = vscode.window.showQuickPick(nodesList, {
+        canPickMany: true,
+        placeHolder: "Please select all the Nodes you want Retina to capture traffic from.",
+        title: "Select Nodes to Capture Traffic From",
+    });
+
+    const selectedNodes = await nodeNamesSelected.then((selectedItems) => {
+        const selectedNodes = selectedItems?.map((item) => item.label).join(",") ?? "";
+        return selectedNodes;
+    });
+
+    if (!selectedNodes) {
+        return;
+    }
+
+    // Retina Run Capture
+    const retinaCaptureResult = await longRunning(`Retina Distributed Capture running for cluster ${clusterInfo.result.name}.`, async () => {
+        return await invokeKubectlCommand(
+            kubectl,
+            kubeConfigFile.filePath,
+            `retina capture create --host-path /mnt/capture --node-selectors "kubernetes.io/os=linux" --node-names "${selectedNodes}" --no-wait=false`,
+        )
+    });
+
+    if (failed(retinaCaptureResult)) {
+        vscode.window.showErrorMessage(`Failed to capture the cluster: ${retinaCaptureResult.error}`);
+        return;
+    }
+
+    if (retinaCaptureResult.result.stderr || retinaCaptureResult.result.code !== 0) {
+        vscode.window.showInformationMessage(`Retina distributed capture failed with following error. ${retinaCaptureResult.result.stdout}`);
+        return;
+    }
+
+    if (retinaCaptureResult.result.stdout && retinaCaptureResult.result.code === 0) {
+        vscode.window.showInformationMessage(`Retina distributed capture is successfully completed for the cluster ${clusterInfo.result.name}`);
+    }
+
+    const kubectlVersion = await getVersion(kubectl, kubeConfigFile.filePath);
+    if (failed(kubectlVersion)) {
+        vscode.window.showErrorMessage(kubectlVersion.error);
+        return;
+    }
+
+    // The pattern of the folder should reflect the pod names and time of the capture
+    const pat = /retina-capture-\w+/g;
+    const match = retinaCaptureResult.result.stdout.match(pat);
+    const startwith = match ? match[0] : null;
```

Comment:
Does it make sense to continue here if there's no match? If that happens, `folderName` will contain the literal string `"null"`.